pkg/stub/retry.go

```diff
@@ -0,0 +1,147 @@
+package stub
+
+import (
+	"fmt"
+	"strings"
+
+	imagev1 "github.com/openshift/api/image/v1"
+	corev1 "k8s.io/api/core/v1"
+	kapis "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// The importTag related functions are for our import image retry and are
+// copied to some degree from the `oc import-image` code, but with various simplifications around options and
+// error detection, as we generally assume the samples imagestreams are valid
+// aside from not needing all the complexity, avoiding having to vendor in openshift/origin is worth
+// some redundancy
+
+func importTag(stream *imagev1.ImageStream, tag string) (*imagev1.ImageStreamImport, error) {
+
+	// follow any referential tags to the destination
+	finalTag, existing, multiple, err := followTagReferenceV1(stream, tag)
+	if err != nil {
+		return nil, err
+	}
+
+	if existing == nil || existing.From == nil {
+		return nil, nil
+	}
+
+	// disallow re-importing anything other than DockerImage
+	if existing.From != nil && existing.From.Kind != "DockerImage" {
+		return nil, fmt.Errorf("tag %q points to existing %s %q, it cannot be re-imported", tag, existing.From.Kind, existing.From.Name)
+	}
+
+	// set the target item to import
+	if multiple {
+		tag = finalTag
+	}
+
+	// reset the generation
+	zero := int64(0)
+	existing.Generation = &zero
```

Comment:
it doesn't look like existing.Generation is used after this point, is this needed?

Reply:
yep some leftover from the borrowed code  ... will remove